{
  "353": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 3,
        "compiler_error_output": "Error: Jai encountered an exception running the program and is terminating.\n\nThis happened outside compile-time execution, so it is likely the compiler's fault.\n\nStack trace of the compiler:\n\n\nStack trace:\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux() [0x74510e]\n/lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7dff6d842520]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN5Typer11infer_typesEP15Ast_Declaration+0x4ca) [0x818aaa]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN5Typer11infer_typesEP6Queued+0x1be) [0x80564e]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux() [0x7cd5e2]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN9Workspace13do_pipe_inferEv+0xff) [0x7cd40f]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_ZN9Workspace10flow_pipesEb+0x11e) [0x7cfe3e]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z13compiler_pumpP9Workspaceb+0x38d) [0x7dc8ad]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z20do_compilation_cycleP9Workspaceb+0x52) [0x7defa2]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z28workspace_thread_entry_pointPv+0x1c0) [0x7df420]\n/home/ubuntu/Desktop/Jai/jai-beta-0.1.096/bin/jai-linux(_Z16thread_middlemanPv+0x35) [0x84a285]\n/lib/x86_64-linux-gnu/libc.so.6(+0x94ac3) [0x7dff6d894ac3]\n/lib/x86_64-linux-gnu/libc.so.6(+0x126850) [0x7dff6d926850]\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 3,
        "compiler_error_output": "Error: Jai encountered an exception running the program and is terminating.\r\n\r\nCommand line arguments:\r\n  C:/jai/bin/jai.exe compiler_bugs/353_352_CEC1.jai\r\n\nThis happened outside compile-time execution, so it is likely the compiler's fault.\n\nStack trace of the compiler:\n\n  unknown(C5E67374) : BaseThreadInitThunk\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/353_352_CEC1.jai",
    "tracking_issue_number": 353
  },
  "356": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\nIn Workspace 2 (\"Target Program\"):\nThe program contains circular dependencies.\nStrangely, we did not find a cycle, so this is probably due to a compiler bug. To help you work around the problem, we'll now print out what is not compiling:\n\nHere are items, waiting to compile, that look suspicious because we don't know why they are waiting:\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: Info: (declaration of 'X')\n\n    X::union{x:X;}\n    ^^^^^^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: s=0x0 type=DECLARATION\n\n    X::union{x:X;}\n    ^^^^^^^^\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: Info: (declaration of 'x')\n\n    X::union{x:X;}\n             ^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: s=0x0 type=DECLARATION\n\n    X::union{x:X;}\n             ^^^\n\n\nHere are some other items, that may just be waiting on the ones listed above:\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: Info: (X)\n\n    X::union{x:X;}\n       ^^^^^\n\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: s=0x0 type=STRUCT\n\n    X::union{x:X;}\n       ^^^^^\n\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\r\nIn Workspace 2 (\"Target Program\"):\r\nThe program contains circular dependencies.\r\nStrangely, we did not find a cycle, so this is probably due to a compiler bug. To help you work around the problem, we'll now print out what is not compiling:\r\n\r\nHere are items, waiting to compile, that look suspicious because we don't know why they are waiting:\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: Info: (declaration of 'X')\r\n\r\n    X::union{x:X;}\r\n    ^^^^^^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,1: s=0x0 type=DECLARATION\r\n\r\n    X::union{x:X;}\r\n    ^^^^^^^^\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: Info: (declaration of 'x')\r\n\r\n    X::union{x:X;}\r\n             ^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,10: s=0x0 type=DECLARATION\r\n\r\n    X::union{x:X;}\r\n             ^^^\r\n\r\n\r\nHere are some other items, that may just be waiting on the ones listed above:\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: Info: (X)\r\n\r\n    X::union{x:X;}\r\n       ^^^^^\r\n\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/356_355_CEC0.jai:1,4: s=0x0 type=STRUCT\r\n\r\n    X::union{x:X;}\r\n       ^^^^^\r\n\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/356_355_CEC0.jai",
    "tracking_issue_number": 356
  },
  "365": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\nIn Workspace 2 (\"Target Program\"):\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,33: Error: Compile-time assertion failed.\n\n    #import \"Basic\";\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\n                                    ^^^^^^^\n\nInfo: In Workspace 2 (\"Target Program\"), building target '365_363_CEC0'.\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Polymorph generation history:\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Info: While generating a polymorph of this procedure.\n\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\n             ^^^^^^^^^^^^^^^^^^^^\n\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: ... with the following constants: (T1=void, T2=Type).\n/home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:6,2: Info: ... The polymorph was triggered here.\n\n    main :: () {\n    \tcheck((cast(*void)0).*, #run (cast(*void)0).*); // OK\n    \tcheck((struct{x:void;}).{}.x, #run (struct{x:void;}).{}.x); // fails\n     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 1,
        "compiler_error_output": "\r\nIn Workspace 2 (\"Target Program\"):\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,33: Error: Compile-time assertion failed.\r\n\r\n    #import \"Basic\";\r\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\r\n                                    ^^^^^^^\r\n\r\nInfo: In Workspace 2 (\"Target Program\"), building target '365_363_CEC0'.\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Polymorph generation history:\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: Info: While generating a polymorph of this procedure.\r\n\r\n    check :: (o1 : $T1, o2 : $T2) { #assert(T1 == T2); }\r\n             ^^^^^^^^^^^^^^^^^^^^\r\n\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:3,10: ... with the following constants: (T1=void, T2=Type).\r\nC:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/365_363_CEC0.jai:6,2: Info: ... The polymorph was triggered here.\r\n\r\n    main :: () {\r\n    \tcheck((cast(*void)0).*, #run (cast(*void)0).*); // OK\r\n    \tcheck((struct{x:void;}).{}.x, #run (struct{x:void;}).{}.x); // fails\r\n     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\r\n\r\n",
        "compiler_output": "",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/365_363_CEC0.jai",
    "tracking_issue_number": 365
  },
  "366": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 0,
        "compiler_error_output": "",
        "compiler_output": "-1\n-2\n-3\n-4\n-5\n-6\n-7\n-8\n-9\n-10\n-11\n-12\n-13\n-14\n-15\n",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 0,
        "compiler_error_output": "",
        "compiler_output": "-1\r\n-2\r\n-3\r\n-4\r\n-5\r\n-6\r\n-7\r\n-8\r\n-9\r\n-10\r\n-11\r\n-12\r\n-13\r\n-14\r\n-15\r\n",
        "expected_compilation_exit_code": 1,
        "expected_run_exit_code": -1,
        "is_runtime_test": false,
        "passed_test": false,
        "run_exit_code": -1,
        "runtime_error_output": "",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/366_362_CEC1.jai",
    "tracking_issue_number": 366
  },
  "368": {
    "beta-0.1.096": {
      "linux": {
        "compilation_exit_code": 0,
        "compiler_error_output": "",
        "compiler_output": "Running linker: /home/ubuntu/Desktop/Jai/jai-beta-0.1.096//bin/lld-linux -flavor Gnu --eh-frame-hdr -export-dynamic -o /home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/368_361_REC0 /home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/.build/368_361_REC0_0_w2.o /lib/x86_64-linux-gnu/crt1.o /lib/x86_64-linux-gnu/crti.o /lib/x86_64-linux-gnu/crtn.o -L /home/ubuntu/Desktop/actions-runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev --dynamic-linker /lib64/ld-linux-x86-64.so.2 -rpath='$ORIGIN' -L /usr/lib/x86_64-linux-gnu/libfakeroot -L /usr/local/lib -L /usr/local/lib/x86_64-linux-gnu -L /lib/x86_64-linux-gnu -L /usr/lib/x86_64-linux-gnu -L /lib64 -L /lib -L /usr/lib64 -L /usr/lib -L /home/ubuntu/Desktop/Jai/jai-beta-0.1.096/modules/ --start-group /lib/x86_64-linux-gnu/libc.so --end-group\n\nStats for Workspace 2 (\"Target Program\"):\nLexer lines processed: 11369 (19072 including blank lines, comments.)\nFront-end time: 0.090547 seconds.\nllvm      time: 0.233856 seconds.\n\nCompiler  time: 0.324403 seconds.\nLink      time: 0.000540 seconds.\nTotal     time: 0.324943 seconds.\n",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": 0,
        "is_runtime_test": true,
        "passed_test": false,
        "run_exit_code": 1,
        "runtime_error_output": "Cast bounds check failed.  Number must be in [0, 255]; it was 1230.  Site is /home/ubuntu/Desktop/Jai/jai-beta-0.1.096/modules/String/module.jai:1227.\nPanic.\nBreakpoint hit. Stack trace:\ncompiler_bugs/368_361_REC0(debug_break_13000000f4+0x29) [0x245359]\ncompiler_bugs/368_361_REC0(my_panic_13000000eb+0x5a) [0x24519a]\ncompiler_bugs/368_361_REC0(__cast_bounds_check_fail_13000000e3+0x5ac) [0x24404c]\ncompiler_bugs/368_361_REC0(parse_int_1700000009+0x1f9) [0x21e319]\ncompiler_bugs/368_361_REC0(__program_main+0x1aa) [0x21b76a]\ncompiler_bugs/368_361_REC0(__system_entry_point_13000000f1+0xa9) [0x248199]\n/lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7e7a14629d90]\n/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7e7a14629e40]\ncompiler_bugs/368_361_REC0(_start+0x25) [0x24b185]\n",
        "runtime_output": ""
      },
      "windows": {
        "compilation_exit_code": 0,
        "compiler_error_output": "",
        "compiler_output": "   Creating library C:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/.build/368_361_REC0.lib and object C:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/.build/368_361_REC0.exp\r\nRunning linker: \"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.39.33519\\bin\\Hostx64\\x64\\link.exe\" /nologo C:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/.build/368_361_REC0_0_w2.obj /OUT:C:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/368_361_REC0.exe /MACHINE:AMD64 /INCREMENTAL:NO /DEBUG /IMPLIB:C:/Users/Private/Desktop/testo_runner/_work/JaiCommunityBugTrackerDev/JaiCommunityBugTrackerDev/compiler_bugs/.build/368_361_REC0.lib /libpath:\"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.39.33519\\lib\\x64\" /libpath:\"C:\\Program Files (x86)\\Windows Kits\\10\\Lib\\10.0.22000.0\\um\\x64\" /libpath:\"C:\\Program Files (x86)\\Windows Kits\\10\\Lib\\10.0.22000.0\\ucrt\\x64\" -nodefaultlib libcmt.lib vcruntime.lib ucrt.lib kernel32.lib kernel32.lib DbgHelp.lib\r\n\r\nStats for Workspace 2 (\"Target Program\"):\r\nLexer lines processed: 10936 (17579 including blank lines, comments.)\r\nFront-end time: 0.080140 seconds.\r\nllvm      time: 0.350789 seconds.\r\n\r\nCompiler  time: 0.430929 seconds.\r\nLink      time: 0.354472 seconds.\r\nTotal     time: 0.785401 seconds.\r\n",
        "expected_compilation_exit_code": 0,
        "expected_run_exit_code": 0,
        "is_runtime_test": true,
        "passed_test": false,
        "run_exit_code": 1,
        "runtime_error_output": "Cast bounds check failed.  Number must be in [0, 255]; it was 1230.  Site is C:/jai/modules/String/module.jai:1227.\nPanic.\nA breakpoint was hit, but no debugger is attached.\n\nPrinting the stack trace:\nhandle_exception                  C:\\jai\\modules\\Runtime_Support_Crash_Handler.jai:361\n... (skipping OS-internal procedures)\ndebug_break                       C:\\jai\\modules\\Runtime_Support.jai:463\nmy_panic                          C:\\jai\\modules\\Runtime_Support.jai:364\n__cast_bounds_check_fail          C:\\jai\\modules\\Runtime_Support.jai:218\nparse_int(T=u8)                   C:\\jai\\modules\\String\\module.jai:1227\nmain                              C:\\Users\\Private\\Desktop\\testo_runner\\_work\\JaiCommunityBugTrackerDev\\JaiCommunityBugTrackerDev\\compiler_bugs\\368_361_REC0.jai:6\n",
        "runtime_output": ""
      }
    },
    "file_path": "compiler_bugs/368_361_REC0.jai",
    "tracking_issue_number": 368
  }
}